src/code/providers/provider-interface.coffee

```diff
@@ -30,6 +30,11 @@ class CloudMetadata
     else
       name
 
+  @newExtension: (name, extension) ->
+    if ~name.indexOf(".")
+      name = name.split(".")[0]
+    name + "." + extension
```

Comment:
In the case of multiple dots in a name, I believe it is customary to assume that the last one indicates the extension rather than the first. A comment under this StackOverflow answer (http://stackoverflow.com/a/4250417) suggests `x.substr(0, x.lastIndexOf('.')) || x`, for instance.